app/models/calagator/source.rb

```diff
@@ -38,10 +38,10 @@ class Source < ActiveRecord::Base
 
   # Create events for this source. Returns the events created. URL must be set
   # for this source for this to work.
-  def create_events!(opts={})
+  def create_events!
     save!
-    events = to_events(opts).select(&:valid?)
-    events.reject!(&:old?) if opts[:skip_old]
+    events = to_events.select(&:valid?)
+    events.reject!(&:old?)
```

Comment:
If it is desirable to avoid repeated enumeration, this could actually be shrunk to:
```ruby
def create_events!
  save!
  to_events.select { |event| event.valid? && !event.old? && event.save! }
end
```

Alternatively:

```ruby
def create_events!
  save!
  to_events.select(&:valid?).reject!(&:old?).each(&:save!)
end
```

Thoughts?